src/sql-parser/src/ast/defs/name.rs

```diff
@@ -66,15 +257,11 @@ impl Ident {
     }
 }
 
-impl From<&str> for Ident {
-    fn from(value: &str) -> Self {
-        Ident(value.to_string())
-    }
-}
-
-impl From<String> for Ident {
-    fn from(value: String) -> Self {
-        Ident(value)
+impl From<SmallString> for Ident {
+    fn from(value: SmallString) -> Self {
+        // Note: using unchecked here is okay because SmallString is known to be less than or equal
+        // to our max length.
```

Comment:
`SmallString` and `Ident` use different constant for their max lengths. They happen to be equal now, but they might diverge in the future by accident. Would it be possible to use the same constant?

Reply:
I updated `SmallString` to be `MaxLenString`, which takes a maximum length as a generic constant. Then updated the `From<...>` impl on `Ident` so we only implement it for `MaxLenString`s with a max len of 255, this should prevent skew in the future.

Now that I think about it, we could probably push down all of these methods onto `MaxLenString`, and then `Ident` becomes a type wrapper around it. If you don't mind I'll probably save this as a followup though?